core/src/main/java/jenkins/model/ModelObjectWithContextMenu.java

```diff
@@ -259,6 +259,12 @@ public void run(JellyContext context, XMLOutput output) throws JellyTagException
          */
         @Exported public boolean requiresConfirmation;
 
+         /**
+         * True to require menu reload after a click.
+         * @since TODO
+         */
+        @Exported public boolean requestsMenuInvalidate;
```

Comment:
Not sure if it's required in REST API, but OK

Reply:
OTOH I would advice against adding new public non-final fields. Better to have a private field and a getter

Reply:
but what about consistency? 
do you think it will be better to refactor all fields of  ```MenuItem``` to private final?
in different PR

Reply:
Refactoring is not trivial, these fields are the part of the public API. I suggest to just avoid growing the technical debt